openhtf/core/__init__.py

```diff
@@ -50,23 +50,27 @@ class TestStopError(Exception):
 class TestExecutor(threads.KillableThread):
   """Encompasses the execution of a single test."""
 
-  def __init__(self, test_descriptor, test_start, teardown_function=None):
+  def __init__(self, test_descriptor, test_start_function, teardown_function=None):
     super(TestExecutor, self).__init__(name='TestExecutorThread')
     self.test_state = None
 
-    # Force teardown function timeout, otherwise we can hang for a long time
-    # when shutting down, such as in response to a SIGINT.
-    timeout_s = conf.teardown_timeout_s
-    if hasattr(teardown_function, 'timeout_s'):
-      timeout_s = teardown_function.timeout_s
+    # Ensure test_start is a phase that saves to the dut_id.
+    if not isinstance(test_start_function, (openhtf.TestPhase, openhtf.PhaseDescriptor)):
+      @openhtf.TestPhase()
+      def trigger_phase(test):
+        test.dut_id = test_start_function()
+    else:
+      trigger_phase = test_start_function
 
-    self._teardown_function = (
-        teardown_function and
-        openhtf.PhaseDescriptor.wrap_or_copy(
-            teardown_function, timeout_s=timeout_s))
+    self._test_start = _sanitize_to_phase(
+        trigger_phase, default_timeout_s=60*60*24*365)
```

Comment:
I'm a little confused by this. `trigger_phase` is either a `TestPhase` or `PhaseDescriptor` at this point. Is the purpose of calling `_sanitize_to_phase` just to ensure that a timeout is set?